yellowbrick/features/pcoords.py

```diff
@@ -169,6 +200,37 @@ def draw(self, X, y, **kwargs):
         Called from the fit method, this method creates the parallel
         coordinates canvas and draws each instance and vertical lines on it.
         """
+        # Convert from dataframe
+        if is_dataframe(X):
+            X = X.as_matrix()
+
+        # Choose a subset of samples
+        if isinstance(self.sample, int):
+            if self.sample < 1:
+                raise YellowbrickValueError(
+                    "`sample` parameter of type `int` must be greater than 1"
+                )
+            n_samples = min([self.sample, len(X)])
+        elif isinstance(self.sample, float):
+            if self.sample <= 0 or self.sample > 1:
+                raise YellowbrickValueError(
+                    "`sample` parameter of type `float` must be between 0 and 1"
+                )
+            n_samples = int(len(X) * self.sample)
+        else:
+            raise YellowbrickTypeError(
+                "`sample` parameter must be int or float"
+            )
+        X = X[:n_samples, :]
+
+        # Normalize
+        if self.normalize is not None:
+            if self.normalize not in self.normalizers:
+                raise YellowbrickValueError(
```

Comment:
This seems a little late for an exception to be raised, and also I'd like to keep the fitted normalizer around in case we need to use `inverse_transform` on it. My suggestion would be to hoist this to `__init__` and instead of keeping self.normalize, actually making it the normalizer as follows:

```python
class ParallelCoordinates(DataVisualizer):

    normalizers = {
        'minmax': MinMaxScalar,
        'minabs': MinAbsScalar, 
        'standard': StandardScalar, 
        'l1': lambda: Normalizer('l1'), 
        'l2': lambda: Normalizer('l2'), 
    }

    def __init__(self, **kwargs):
        ....

        if normalize and normalize not in self.normalizers:
            raise YellowbrickValueError() 
        self.normalize = self.normalizers.get(normalize, None) 

    def fit(self, X, y):
        if self.normalize:
            self.normalize.fit(X) 
        # maybe also compute indices of the sample 

    def transform(self, X):
        # maybe also perform sampling 
        if self.normalize:
            return self.normalize.transform(X) 
        return X
```

Of course, we need to figure out where `draw()` is called to make this work.